src/androidTest/java/androidx/widget/TextViewTest.kt

```diff
@@ -0,0 +1,184 @@
+
+        assertEquals(drawableRed, textView.compoundDrawablesRelative[0])
+        assertEquals(drawableRed, textView.compoundDrawablesRelative[2])
+        assertEquals(drawableRed, textView.compoundDrawablesRelative[3])
+        assertNull(textView.compoundDrawablesRelative[1])
+    }
+
+    @Test
+    fun updateCompoundDrawablesRelative_keepsDrawables() {
+        val drawableRed = ColorDrawable(Color.RED)
+        val drawableBlack = ColorDrawable(Color.BLACK)
+        textView.updateCompoundDrawablesRelative(
+                start = drawableRed, bottom = drawableRed, end = drawableRed)
+        textView.updateCompoundDrawablesRelative(
+                start = drawableBlack, top = drawableRed, end = null)
+
+        assertEquals(drawableBlack, textView.compoundDrawablesRelative[0])
+        assertEquals(drawableRed, textView.compoundDrawablesRelative[1])
+        assertEquals(drawableRed, textView.compoundDrawablesRelative[3])
+        assertNull(textView.compoundDrawablesRelative[2])
+    }
+
+    @Test
+    fun updateCompoundDrawablesRelativeWithIntrinsicBounds() {
+        val drawable = ColorDrawable(Color.RED)
+        textView.updateCompoundDrawablesRelativeWithIntrinsicBounds(top = drawable)
+
+        assertEquals(drawable, textView.compoundDrawablesRelative[1])
+        assertNull(textView.compoundDrawablesRelative[0])
+        assertNull(textView.compoundDrawablesRelative[2])
+        assertNull(textView.compoundDrawablesRelative[3])
+    }
+
+    @Test
+    fun updateCompoundDrawablesRelativeWithIntrinsicBounds_keepsDrawables() {
+        val drawableRed = ColorDrawable(Color.RED)
+        val drawableBlack = ColorDrawable(Color.BLACK)
+        textView.updateCompoundDrawablesRelativeWithIntrinsicBounds(
+                top = drawableRed, bottom = drawableRed)
+        textView.updateCompoundDrawablesRelativeWithIntrinsicBounds(
+                bottom = null, end = drawableBlack)
+
+        assertEquals(drawableRed, textView.compoundDrawablesRelative[1])
+        assertEquals(drawableBlack, textView.compoundDrawablesRelative[2])
+        assertNull(textView.compoundDrawablesRelative[0])
+        assertNull(textView.compoundDrawablesRelative[3])
+    }
+
+    @Test
+    fun updateCompoundDrawablesRelativeWithIntrinsicBoundsWithResourceIds() {
+        textView.updateCompoundDrawablesRelativeWithIntrinsicBounds(
+                start = android.R.drawable.btn_default,
+                top = android.R.drawable.btn_default,
+                bottom = android.R.drawable.btn_default,
+                end = android.R.drawable.btn_default)
+
+        assertNotNull(textView.compoundDrawablesRelative[0])
+        assertNotNull(textView.compoundDrawablesRelative[1])
+        assertNotNull(textView.compoundDrawablesRelative[2])
+        assertNotNull(textView.compoundDrawablesRelative[3])
+    }
+
+    @Test
+    fun updateCompoundDrawablesRelativeWithIntrinsicBoundsWithResourceIds_keepsDrawables() {
```

Comment:
Don't you think adding tests on forbidden values would be useful? Like -2?

Reply:
Not a bad idea

Reply:
Thought it could be better to demand correct argument with a `@IntRange` floor value requirement.  Tests added too. Thoughts?
```kotlin
@DrawableRes
const val UPDATE_MODE_REMOVE = 0
@DrawableRes
const val UPDATE_MODE_KEEP = -1
// ...
fun TextView.updateCompoundDrawablesWithIntrinsicBounds(
    @DrawableRes @IntRange(from = UPDATE_MODE_KEEP.toLong()) left: Int = UPDATE_MODE_KEEP,
// ...
```

Reply:
Are you sure this'll work in all cases?
https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/content/res/ResourceId.java#33